### PR TITLE
Move global shader variables into new OceanInput

### PR DESCRIPTION
**Pull request by daleeidd**
This is to converge with changes introduced downstream. Created OceanInput.hlsl and moved global shader variables into it. Also rearranged the includes on Ocean.shader to match downstream.

### CODE REVIEW

File: crest/Assets/Crest/Crest/Shaders/OceanInput.hlsl [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+half _SubSurfaceBase;
+half _SubSurfaceSun;
+half _SubSurfaceSunFallOff;
+
+half _SubSurfaceDepthMax;
+half _SubSurfaceDepthPower;
+half3 _SubSurfaceShallowCol;
+half3 _SubSurfaceShallowColShadow;
+
+half _CausticsTextureScale;
```
Comment (huwb) [inline crest/Assets/Crest/Crest/Shaders/OceanInput.hlsl:42]: should these be behind #ifdef CAUSTICS , and similar questions for other params that are behind shader options? maybe its fine as it is - maybe unity removes unused ones anyway..

**Review by huwb — CHANGES_REQUESTED** [review]
Happy to see this change, good stuff

My only request would be could we find a different name than OceanInput, as this combination of words is heavily used in crest for ocean inputs. I see a file LitData.hlsl in SRP that serves this kind of purpose. Maybe OceanShaderData.hlsl? (If you do use this name then consider it approved from my POV).

Thanks!

File: crest/Assets/Crest/Crest/Shaders/OceanInput.hlsl (hunk 1 above)
Comment (daleeidd) [reply crest/Assets/Crest/Crest/Shaders/OceanInput.hlsl:42]: I think they do. I was just matching what we have downstream. I will add them in here and downstream.

**Comment by daleeidd** [comment]
> What are the downstream changes? I think i can make a pretty good guess but just curious.

- Global shader variables are in their own file
- The includes are grouped before the vert function in Ocean.shader
- Less ifdefs (I will add them back in and add them downstream instead)
- Some includes are cleaned up like in Underwater Curtain

> My only request would be could we find a different name than OceanInput, as this combination of words is heavily used in crest for ocean inputs. I see a file LitData.hlsl in SRP that serves this kind of purpose. Maybe OceanShaderData.hlsl? (If you do use this name then consider it approved from my POV).

Can do.

**Comment by daleeidd** [comment]
Done all of those. I have reorganised the file and added some headers.

**Review by huwb — APPROVED** [review]
Looks good

**Comment by daleeidd** [comment]
Thank you

**Pull request merged**